src/taipy/_cli/_migrate.py

```diff
@@ -0,0 +1,483 @@
+    parents = []
+    possible_parents = "TASK" if entity_type == "DATA_NODE" else "SCENARIO"
+    data = config[possible_parents]
+
+    for _id, entity_data in data.items():
+        section_id = f"{entity_id}:SECTION"
+        if entity_type == "DATANODE" and possible_parents == "TASK":
+            if section_id in entity_data["input_ids"] or section_id in entity_data["output_ids"]:
+                parents.append(section_id)
+
+        if entity_type == "TASK" and possible_parents == "SCENARIO":
+            if section_id in entity_data["tasks"]:
+                parents.append(section_id)
+    return parents
+
+
+def __fetch_tasks_from_pipelines(pipelines: List, data: Dict) -> List:
+    tasks = []
+    for pipeline in pipelines:
+        pipeline_data = data[pipeline]["data"]
+        tasks.extend(pipeline_data["tasks"])
+    return tasks
+
+
+def _migrate_scenario(scenario: Dict, data: Dict) -> Dict:
+    # pipelines were replaced by tasks
+    scenario["tasks"] = __fetch_tasks_from_pipelines(scenario["pipelines"], data)
+
+    # pipeline attribute not removed in 3.0
+    scenario["pipelines"] = None
+
+    # additional_data_nodes attribute added
+    scenario["additional_data_nodes"] = None
+
+    return scenario
+
+
+def _migrate_task(task: Dict) -> Dict:
+    # owner_id was not present in 2.0
+    if not task.get("owner_id"):
+        task["owner_id"] = task["parent_ids"][0] if task["parent_ids"] else None
+
+    # properties was not present in 2.0
+    task["properties"] = task.get("properties", {})
+
+    # skippable was not present in 2.0
+    task["skippable"] = task.get("skippable", False)
+
+    return task
+
+
+def _migrate_task_entity(task: Dict, data: Dict) -> Dict:
+    task = __update_parent_ids(task, data)
+    return _migrate_task(task)
+
+
+def _migrate_task_config(id: str, task: Dict, config: Dict) -> Dict:
+    task = __update_config_parent_ids(id, task, "TASK", config)
+    return _migrate_task(task)
+
+
+def _migrate_datanode(datanode: Dict) -> Dict:
+    # cacheable was removed in after 2.0
+    _ = datanode.pop("cacheable", False)
+
+    # job_ids was replaced by edits
+    if "job_ids" in datanode:
+        datanode["edits"] = [{"job_id": job, "timestamp": datanode["last_edit_date"]} for job in datanode["job_ids"]]
+    elif "edits" in datanode:
+        # make sure timestamp inside edits is a string
+        edits = []
+        for edit in datanode["edits"]:
+            timestamp = edit.get("timestamp")
+            if isinstance(timestamp, dict):
+                timestamp = timestamp.get("__value__")
+            new_edit = {"timestamp": timestamp}
+            if "job_id" in edit:
+                new_edit["job_id"] = edit["job_id"]
+            edits.append(new_edit)
+        datanode["edits"] = edits
+
+    # owner_id was not present in 2.0, need to be search for in tasks
+    datanode["owner_id"] = datanode.get("owner_id", None)
```

Comment:
```suggestion
    datanode["owner_id"] = datanode.get("parent_id", None)
```
This line should migrate "parent_id" to "owner_id". Can you check again?
Also, I believe we need to pop the parent_id if it exists after migrating to owner_id